src/world_generation/perlin.rs

```diff
@@ -1,19 +1,63 @@
+use rand::distributions::Open01;
 /// We need a Perlin noise generator, with a few characteristics:
-/// 
+///
 /// - it is not a one shot thing: we must be able to generate on demand more
 ///   noise
 /// - it must be infinite
 /// - we must be able to spawn a few different instances
 /// - we must be able to change the frequency of the noise
 /// - it must be **fully** deterministic, in the sense that generating noise in
 ///   a different order will not change the outcome
-
 use rand::rngs::SmallRng;
 use rand::{Rng, SeedableRng};
-use rand::distributions::Open01;
-use std::iter::zip;
 use std::collections::HashMap;
+use std::iter::zip;
+
+/// Number of scales to use in the Perlin Noise
+const SCALE_LEVEL_PERLIN: usize = 5;
+
+/// Class containing the different scales of Perlin noise,
+/// combines them to return a single value for each querried coord.
+pub struct MultiscalePerlinNoise {
+    perlin_noises: Vec<PerlinNoise>,
+    amplitudes: [f32; SCALE_LEVEL_PERLIN],
```

Comment:
Move `amplitude`  inside `PerlinNoise`

Reply:
Then the constructor receives simple 

`new(perlin_noises: &[PerlinNoise])` 

using `&[T]` means "refernce to a slice", which can be either 
- an array: &[t2, t3, t3]
- a vec: &vec![t1,t2,t3]

Then use `perlin_noises.to_vec()` to cast it as a Vec